cognite/pygen/utils/helper.py

```diff
@@ -21,3 +21,35 @@ def chdir(new_dir: Path) -> Iterator[None]:
 
     finally:
         os.chdir(current_working_dir)
+
+
+def _get_literal_str(text: str) -> str | None:
+    """Finds the Literal in a string and returns it.
+
+    Args:
+        text: The string to search for the Literal.
+
+    Returns:
+        The Literal in the string, or None if not found.
+
+    Examples:
+
+        >>> _get_literal_str("Literal['hello']")
+        "Literal['hello']"
+        >>> _get_literal_str("str | int")
+
+        >>> _get_literal_str("Literal['hello', 'world'] | int")
+        "Literal['hello', 'world']"
+        >>> _get_literal_str('None | Literal["hello", "world", "!"]')
+        'Literal["hello", "world", "!"]'
+    """
+    if "Literal[" not in text:
+        return None
+
+    start = text.find("Literal[")
+    end = text.find("]", start) + 1
+
+    if start == -1 or end == -1:
```

Comment:
How can `end` be `-1` when you add `+ 1` above?

Reply:
Good catch. Added extra test and fixed the issue.